enhancements/network/configurable-network-diagnostics-pod-placement.md

```diff
@@ -0,0 +1,303 @@
+pod has to be scheduled only on a specific subset of nodes.
+
+An additional requirement is to handle a scenario where no nodes are available to deploy the 
+`network-check-source` pod. Instead of the CNO remaining in a `Progressing` state waiting for the 
+deployment to be scheduled, it should detect and handle the use case without affecting the 
+overall operator state.
+
+### User Stories
+
+* As an OpenShift cluster administrator, I want to ensure that the `network-check-source` pod is 
+  only scheduled on an infrastructure node.
+* As an OpenShift cluster administrator, I want to control the set of nodes 
+  `network-check-target` and `network-check-source` pods are scheduled on.
+* As an OpenShift cluster administrator, I want to scale the worker nodes down to zero without 
+  permanently affecting the network operator conditions.
+  
+### Goals
+
+* Enable OpenShift cluster administrators to control the pod placement of the 
+  `network-check-source` Deployment and the `network-check-target` DaemonSet.
+* Ensure that CNO handles a scenario where there are no available nodes for the 
+  `network-check-source` Deployment without permanently affecting the operator conditions.
+
+### Non-Goals
+
+* Modifying the scope of the connectivity checks performed by the network diagnostics feature.
+* Modifying other properties of the `network-check-source` Deployment and the 
+  `network-check-target` DaemonSet.
+
+## Proposal
+
+This enhancement adds new API fields to the `network.config.openshift.io` Custom Resource 
+Definition (CRD), that allow administrators to control the pod placement of the 
+`network-check-source` Deployment and the `network-check-target` DaemonSet. During the 
+reconciliation process for the mentioned resources, CNO will configure the relevant 
+`tolerations` and `nodeSelector` fields.  
+Additionally, introduce the `NetworkDiagnosticsAvailable` status condition that's going to reflect
+whether network diagnotics are currently available. This is useful in a scenario where there are 
+no nodes that can host the `network-check-source` Deployment. CNO will set the condition 
+instead of putting the operator into a constant `Progressing` state. Considering that the 
+network diagnostics feature is only informational it should not affect the `Available` and 
+`Degraded` conditions.
+
+### Workflow Description
+
+The OpenShift cluster administrator can configure the pod placement of the 
+`network-check-source` Deployment and the `network-check-target` DaemonSet by changing the `.
+spec.networkDiagnostics.sourcePlacement` and `.spec.networkDiagnostics.targetPlacement` fields 
+in the `cluster/network.config.openshift.io` object.  
+
+1. The OpenShift cluster administrator modifies the `network.config.openshift.io` custom 
+   resource, for example:
+   ```
+   apiVersion: config.openshift.io/v1
+   kind: Network
+   spec:
+     ...
+     networkDiagnostics:
+       mode: All
```

Comment:
What does this `mode: All` mean?

Reply:
It seems related to enable/disable network diagnostics, shall we can manually set it to `Disabled`, if yes, what will happen ?  No `network-check-source` pods and the `network-check-target` pods ?